### PR TITLE
CLD-348: Configure swap space recommendations

### PR DESCRIPTION
**Pull request by sumanthravipati**

### CODE REVIEW

**Review by vitalykorolev — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -148,6 +148,21 @@ local     1b65575a84be319222a4ff9ba9eecdff06ffb3143edbd03720f4b808be0e6d18
 local     MarkLogic_vol_1
 ```
 
+## Configuring swap space
+
+MarkLogic recommends that swap space be configured for production deployments to reduce the possibility of ‘out of memory’ errors. Please find more details on [MarkLogic recommendations for swap space](https://help.marklogic.com/knowledgebase/article/View/21/19/swap-space-requirements) and [configuring "swappiness"](https://help.marklogic.com/Knowledgebase/Article/View/linux-swappiness).
+
+Check on how [Docker allocates swap space](https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details). The amount of memory and swap that are available to MarkLogic can be controlled using the "--memory" and "--memory-swap" settings. For example, if you wanted to run a MarkLogic container with you 64GB of memory and 32GB of swap, you would specify the following with your docker run command:
```
Comment (jkerr5) [inline README.md:155]: Suggest some rewording: 

In Docker, the amount of memory and swap that are available to MarkLogic can be controlled using the "--memory" and "--memory-swap" settings. See the Docker documentation for [--memory-swap-details](https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details) for details.

**Review by jkerr5 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -148,6 +148,21 @@ local     1b65575a84be319222a4ff9ba9eecdff06ffb3143edbd03720f4b808be0e6d18
 local     MarkLogic_vol_1
 ```
 
+## Configuring swap space
+
+MarkLogic recommends that swap space be configured for production deployments to reduce the possibility of ‘out of memory’ errors. Please find more details on [MarkLogic recommendations for swap space](https://help.marklogic.com/knowledgebase/article/View/21/19/swap-space-requirements) and [configuring "swappiness"](https://help.marklogic.com/Knowledgebase/Article/View/linux-swappiness).
```
Comment (jmckean17) [inline README.md:153]: Sumanth, make the second sentence "For more details, see...."

File: README.md [hunk 3]
```diff
@@ -148,6 +148,21 @@ local     1b65575a84be319222a4ff9ba9eecdff06ffb3143edbd03720f4b808be0e6d18
...
+
+In Docker, the amount of memory and swap that are available to MarkLogic can be controlled using the "--memory" and "--memory-swap" settings. See the Docker documentation [--memory-swap-details](https://docs.docker.com/config/containers/resource_constraints/#--memory-swap-details) for more details. For example, if you want to run a MarkLogic container with 64GB of memory and 32GB of swap, you would specify the following with your docker run command:
```
Comment (jmckean17) [inline README.md:155]: ...."memory and swap space that are ..."

File: README.md [hunk 4]
```diff
@@ -148,6 +148,21 @@ local     1b65575a84be319222a4ff9ba9eecdff06ffb3143edbd03720f4b808be0e6d18
...
+```
+--memory="64g" --memory-swap="96g"
+```
+If you want to limit memory to 64GB but allow MarkLogic to use unlimited swap, up to the amount available on the host system, specify the following with your docker run command:
```
Comment (jmckean17) [inline README.md:159]: MarkLogic Server
swap space
(up to the amount available on the host system),

File: README.md [hunk 5]
```diff
@@ -148,6 +148,21 @@ local     1b65575a84be319222a4ff9ba9eecdff06ffb3143edbd03720f4b808be0e6d18
...
+```
+--memory="64g" --memory-swap="-1"
+```
+To allow MarkLogic to be able to use unlimited memory and swap, up to the amount available on the host system, do not specify either "--memory" or "--memory-swap".
```
Comment (jmckean17) [inline README.md:163]: "To allow MarkLogic Server to use unlimited memory and swap (up to the amount available on the host system), do not...."

**Review by jmckean17 — COMMENTED** [review]
Sumanth, will this be the content for Docker GA? There's a note about Docker EA images in the text. 

Let me know if you have questions. Thanks, --Jane

**Review by barkhachoithani — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (sumanthravipati) [reply README.md]: Updated.

File: README.md (hunk 3 above)
Comment (sumanthravipati) [reply README.md]: Updated

File: README.md (hunk 4 above)
Comment (sumanthravipati) [reply README.md]: Updated

File: README.md (hunk 5 above)
Comment (sumanthravipati) [reply README.md]: Updated

**Pull request merged**